extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ClickDelayFilter, makeClickDelaySpec} from './filters/click-delay';
+import {FilterType} from './filters/filter';
+import {assertConfig, TransportMode} from './config';
+import {isExperimentOn} from '../../../src/experiments';
+import {isJsonScriptTag, openWindowDialog} from '../../../src/dom';
+import {urlReplacementsForDoc} from '../../../src/services';
+import {user} from '../../../src/log';
+
+const TAG = 'amp-ad-exit';
+
+export class AmpAdExit extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!./config.AmpAdExitConfig} */
+    this.config_ = {targets: {}, filters: {}};
```

Comment:
It was convenient to use the parsed JSON. With @lannka's suggested refactoring this isn't useful anymore.